packages/browser/__tests__/Session.spec.ts

```diff
@@ -0,0 +1,205 @@
+    });
+  });
+
+  describe("fetch", () => {
+    it("wraps up ClientAuthentication fetch", async () => {
+      const clientAuthentication = mockClientAuthentication();
+      const clientAuthnFetch = jest.spyOn(clientAuthentication, "fetch");
+      const mySession = new Session({ clientAuthentication });
+      await mySession.fetch("https://some.url");
+      expect(clientAuthnFetch).toHaveBeenCalled();
+    });
+  });
+
+  describe("handleincomingRedirect", () => {
+    it("wraps up ClientAuthentication handleIncomingRedirect", async () => {
+      const clientAuthentication = mockClientAuthentication();
+      const clientAuthnHandle = jest.spyOn(
+        clientAuthentication,
+        "handleIncomingRedirect"
+      );
+      const mySession = new Session({ clientAuthentication });
+      await mySession.handleIncomingRedirect("https://some.url");
+      expect(clientAuthnHandle).toHaveBeenCalled();
+    });
+
+    it("updates the session's info if relevant", async () => {
+      const clientAuthentication = mockClientAuthentication();
+      clientAuthentication.handleIncomingRedirect = jest.fn(
+        async (_url: string) => {
+          return {
+            isLoggedIn: true,
+            sessionId: "a session ID",
+            webId: "https://some.webid#them",
+          };
+        }
+      );
+      const mySession = new Session({ clientAuthentication });
+      expect(mySession.info.isLoggedIn).toEqual(false);
+      await mySession.handleIncomingRedirect("https://some.url");
+      expect(mySession.info.isLoggedIn).toEqual(true);
+      expect(mySession.info.sessionId).toEqual("a session ID");
+      expect(mySession.info.webId).toEqual("https://some.webid#them");
+    });
+
+    it("leaves the session's info unchanged if no session is obtained after redirect", async () => {
+      const clientAuthentication = mockClientAuthentication();
+      clientAuthentication.handleIncomingRedirect = jest.fn(
+        async (_url: string) => undefined
+      );
+      const mySession = new Session({ clientAuthentication }, "mySession");
+      await mySession.handleIncomingRedirect("https://some.url");
+      expect(mySession.info.isLoggedIn).toEqual(false);
+      expect(mySession.info.sessionId).toEqual("mySession");
+    });
+  });
+
+  describe("onLogin", () => {
+    it("calls the registered callback on login", async (done) => {
+      const myCallback = (): void => {
+        if (done) {
+          done();
+        }
+      };
+      const clientAuthentication = mockClientAuthentication();
+      clientAuthentication.handleIncomingRedirect = jest.fn(
+        async (_url: string) => {
+          return {
+            isLoggedIn: true,
+            sessionId: "a session ID",
+            webId: "https://some.webid#them",
+          };
+        }
+      );
+      const mySession = new Session({ clientAuthentication });
+      mySession.onLogin(myCallback);
+      await mySession.handleIncomingRedirect("https://some.url");
+    });
```

Comment:
No assertion here - could you just set a simple boolean flag in the `myCallback` function and then assert that it's set here...?

Reply:
The assertion is that the `done()` function gets called: if not, the test will time out.

Reply:
Yeah I know, I got that. But would a simple boolean flag just be clearer (and also I'm kinda surprised that our linter doesn't complain about the lack of an explicit assertion in this test (or is it just clever enough to recognise the use of `done()`)...?

Reply:
Yeah, I was also quite surprised that the linter didn't complain. Looking into it, it seems like there's an issue with the linter config: it just doesn't detect empty tests. I installed locally `eslint-plugin-jest`, but that didn't make a difference. I'll investigate as part of a separate issue.

Reply:
Added an explicit expect in 92e8ae0b14dc6a0577a72838a2cb4bf306fffe85